mlog/log.go

```diff
@@ -187,3 +189,10 @@ func (l *Logger) Error(message string, fields ...Field) {
 func (l *Logger) Critical(message string, fields ...Field) {
 	l.zap.Error(message, fields...)
 }
+
+func (l *Logger) SetBuffer(buffer *bytes.Buffer) {
```

Comment:
Instead of extending `Logger` to provide access to this inside tests, would it make sense to construct the buffer inside `setupTestHelper`, pass it /to/ `NewTestingLogger`, and store the buffer on the `TestHelper` struct? This would eliminate any changes to the "production" code. We would then go a step further and leverage an [`io.MultiWriter`](https://golang.org/pkg/io/#MultiWriter) when calling `SetLogger` from `setupTestHelper` so that even the `testingWriter` can remain unchanged, fully isolating these changes to the test helper.

Thoughts?

Reply:
Way better solution! Thanks @lieut-data !